flink-cdc-connect/flink-cdc-source-connectors/flink-cdc-base/src/main/java/org/apache/flink/cdc/connectors/base/relational/connection/JdbcConnectionPools.java

```diff
@@ -83,4 +84,12 @@ public String getJdbcUrl(
         }
         return jdbcConnectionPoolFactory.getJdbcUrl(sourceConfig);
     }
+
+    public static void close() throws IOException {
```

Comment:
Should we use `synchronized` on this method?

Reply:
Good idea, I add pools as synchronized lock( same as JdbcConnectionPools#getOrCreateConnectionPool) to protect operations to get，put , or remove pools.